sycl/source/detail/filter_selector_impl.cpp

```diff
@@ -129,6 +131,7 @@ int filter_selector_impl::operator()(const device &Dev) const {
       else
         DeviceTypeOK = (DT == Filter.DeviceType);
     }
+#ifndef __INTEL_PREVIEW_BREAKING_CHANGES
```

Comment:
Can you clarify why this code relates only to one path and not the other? It's not immediately clear from the variable names below...

Reply:
I've fixed this. Apparently, removing this branch was changing the functionality of filter selector. So, instead I've now added a variable "MatchesSeen" in `ods_target` to keep track of the devices that matches this filter.